src/_pytest/fixtures.py

```diff
@@ -276,24 +277,8 @@ def reorder_items_atscope(items, argkeys_cache, items_by_argkey, scopenum):
 
 def fillfixtures(function):
     """ fill missing funcargs for a test function. """
-    try:
-        request = function._request
-    except AttributeError:
-        # XXX this special code path is only expected to execute
-        # with the oejskit plugin.  It uses classes with funcargs
-        # and we thus have to work a bit to allow this.
-        fm = function.session._fixturemanager
-        fi = fm.getfixtureinfo(function.parent, function.obj, None)
-        function._fixtureinfo = fi
-        request = function._request = FixtureRequest(function)
-        request._fillfixtures()
-        # prune out funcargs for jstests
-        newfuncargs = {}
-        for name in fi.argnames:
-            newfuncargs[name] = function.funcargs[name]
-        function.funcargs = newfuncargs
-    else:
-        request._fillfixtures()
+    warnings.warn(FILLFUNCARGS, stacklevel=2)
+    function._request._fillfixtures()
```

Comment:
strictly speaking this makes it a breaking change already, so direct removal may be advisable

Reply:
Deprecation might help those who used this as just an alias to `_fillfixtures`.

Anyway, since it is public `pytest.*` API I would fill bad just removing it; however that compat functionality it had is almost certainly useless by now.

Reply:
if the start is a deprecation, i believe its necessary to leave the compat code in until the function is removed even if we believe there are practically no users, as it constitutes a breaking change to take away the extra behavior

alternatively, i we do a braking change anyway, the complete one should be done

Reply:
OK, I'll close this PR and open a new one with just the deprecation, hopefully to be removed in pytest 6.0.